src/libostree/ostree-repo-pull.c

```diff
@@ -1523,6 +1559,30 @@ scan_one_metadata_object_c (OtPullData         *pull_data,
       is_stored = TRUE;
       is_requested = TRUE;
     }
+  /* Do we have any --reference local repos? */
+  else if (!is_stored && pull_data->localcache_repos)
+    {
+      for (guint i = 0; i < pull_data->localcache_repos->len; i++)
+        {
+          OstreeRepo *refd_repo = pull_data->localcache_repos->pdata[i];
+          gboolean localcache_repo_has_obj;
+
+          if (!ostree_repo_has_object (refd_repo, objtype, tmp_checksum,
+                                       &localcache_repo_has_obj, cancellable, error))
+            return FALSE;
+          if (!localcache_repo_has_obj)
+            continue;
+          if (!ostree_repo_import_object_from_with_trust (pull_data->repo, refd_repo,
+                                                          objtype, tmp_checksum,
+                                                          !pull_data->is_untrusted,
+                                                          cancellable, error))
+            return FALSE;
+          is_stored = TRUE;
+          is_requested = TRUE;
+          pull_data->n_fetched_localcache_metadata++;
```

Comment:
I was going to suggest adding:

```c
if (objtype == OSTREE_OBJECT_TYPE_COMMIT)
  g_hash_table_add (pull_data->fetched_detached_metadata, g_strdup (tmp_checksum));
```

here as well... but I think it's better without it, right?

It seems safer to always fetch it from the actual remote. Not to mention, we might not be embedding detached metadata in the local cache repo for some reason (like in the Fedora ISO).